File: app/src/main/java/com/example/cb/test/mvp/presenter/NewListPresenter.java
package com.example.cb.test.mvp.presenter;

import com.example.cb.test.mvp.BasePresenter;
import com.example.cb.test.mvp.presenter.view.INewListView;
import com.example.cb.test.rx.http.MovieHttpRequest;
import com.example.cb.test.rx.http.NewsResp;
import com.example.cb.test.rx.http.XHttpCallback;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by bin on 2018/12/24.
 */
public class NewListPresenter extends BasePresenter<INewListView> {

    public NewListPresenter(INewListView view) {
        super(view);
    }

    public void fetchData() {
        Map<String, String> map = new HashMap<>();
        map.put("type", "top");
        map.put("key", "f323c09a114635eb935ed8dd19f7284e");
        MovieHttpRequest.getInstance().sendNewsRequest(map, new XHttpCallback<NewsResp>() {
            @Override
            public void onSuccess(NewsResp newsResp) {
                mView.onRequestSuccess(newsResp);
            }

            @Override
            public void onError(String error) {
                mView.onError(error);
            }
        });
    }
}

File: app/src/main/java/com/example/cb/test/mvp/presenter/NewListPresenter.kt
package com.example.cb.test.mvp.presenter

import com.example.cb.test.base.BasePresenter
import com.example.cb.test.mvp.view.INewListView
import com.example.cb.test.rx.http.MovieHttpRequest
import com.example.cb.test.rx.http.NewsResp
import com.example.cb.test.rx.http.XHttpCallback

/**
 * ====================================================
 * @User :caobin
 * @Date :2019/7/29 10:49
 * @Desc :逻辑处理类
 * ====================================================
 */
class NewListPresenter(view: INewListView) : BasePresenter<INewListView>(view) {

    fun fetchData() {
        val map = HashMap<String, String>()
        map["type"] = "top"
        map["key"] = "f323c09a114635eb935ed8dd19f7284e"
        MovieHttpRequest.getInstance().sendNewsRequest(map, object : XHttpCallback<NewsResp> {
            override fun onSuccess(t: NewsResp?) {
                if (t != null) {
                    mView?.onRequestSuccess(t)
                }
            }

            override fun onError(error: String?) {
                if (error != null) {
                    mView?.onError(error)
                }
            }
        })
    }
}